pkg/client/mapper.go

```diff
@@ -103,3 +103,57 @@ func NewBidirectionalNewOneToManyGroupMapper(source, dest, groupMappingFile stri
 	}
 	return nil, nil, fmt.Errorf("unsupported source to dest mapper type: source %s, dest %s", source, dest)
 }
+
+// UserMapperImpl implements groupsync.UserMapper.
+type UserMapperImpl map[string]string
```

Comment:
Two things:

1. I strongly recommend making this a struct with a private field for the map. It will make future refactoring much simpler because you don't leak the type to the user.

1. Things named `Impl` is almost always an anti-pattern. Why this name?

Reply:
Naming this as `Impl` was because the interface is called `UserMapper`, I think naming this as `UserMapper` is confusing and I want this to be a generic Mapper for call systems.

Reply:
1. Made `map[strng][string]` a private field.
2. Changed the name to `UserMapper`

Reply:
It seems like you're going to want a separate package for each mapper.

Reply:
Discussed offline, created different package for different mapper.